nanome_vault/VaultServer.py

```diff
@@ -59,6 +59,36 @@ def _parse_path(self):
         except:
             Logs.error('Error parsing path:\n', traceback.format_exc())
 
+    def _is_auth_valid(self, path):
+        if not VaultServer.enable_auth:
+            return True
+
+        auth = self.headers.get('authorization')
+        if auth is None:
+            return False
+
+        token = auth.split(' ')[-1]
+        cached = VaultServer.auth_cache.get(token, None)
+
+        if cached is None:
+            headers = {'authorization': auth}
+            r = requests.get('https://api.nanome.ai/user/session', headers=headers)
+            if r.status_code == 200:
```

Comment:
Could be a good idea to display something like "Can't join Nanome API, please try again later" in case of timeout or server error

Reply:
Indeed a good idea, but low priority. For now, if request errors or times out, VaultServer will respond with 401.